zerver/views/invite.py

```diff
@@ -232,6 +233,58 @@ def resend_user_invite_email(
     return json_success(request)
 
 
+@require_member_or_admin
+@typed_endpoint
+def edit_multiuse_invite(
+    request: HttpRequest,
+    user_profile: UserProfile,
+    *,
+    invite_id: PathOnly[int],
+    stream_ids: Json[list[int]],
+    invite_as: Annotated[
+        Json[int],
+        check_int_in_validator(list(PreregistrationUser.INVITE_AS.values())),
+    ],
+    include_realm_default_subscriptions: Json[bool] = False,
+) -> HttpResponse:
+    try:
+        invite = MultiuseInvite.objects.get(id=invite_id)
+    except MultiuseInvite.DoesNotExist:
+        raise JsonableError(
+            _("Invite does not exist with id: {invite_id}.").format(invite_id=invite_id)
+        )
+
+    if not user_profile.can_create_multiuse_invite_to_realm():
+        # Guest users case will not be handled here as it will
+        # be handled by the decorator above.
+        raise JsonableError(_("Insufficient permission"))
```

Comment:
Do we want to allow to edit invites created by other user to non-admins? I don't think we should.